### PR TITLE
fixes #23124 - validates services on prior to backup start

### PR DESCRIPTION
**Pull request by cfouant**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.17
* [ ] 1.16
* [ ] 1.15

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---
To test:
- Run `# katello-service stop --exclude goferd,foreman-proxy,squid,smart_proxy_dynflow_core,qdrouterd,qpidd`
- Run `# katello-backup --online-backup`

Backup should be created without failure

### CODE REVIEW

File: packages/katello/katello/backup.rb [hunk 1]
```diff
@@ -372,6 +371,11 @@ def validate_directory
       end
     end
 
+    def validate_services
+      result = run_cmd("katello-service status #{@excluded}", [0,1])
+      result.include?('dead') ? false : true
```
Comment (johnpmitsch) [inline packages/katello/katello/backup.rb]: `result.include?('dead')` should return true/false on its own

Comment (cfouant) [reply packages/katello/katello/backup.rb]: Yes, but this is backwards. So if services are dead, that means the services are down and thus validate_services would be false.

Comment (cfouant) [reply packages/katello/katello/backup.rb]: I could negate it in one line, but I believe this is the chosen format rather than `not` or `!`

Comment (johnpmitsch) [reply packages/katello/katello/backup.rb]: I've never seen this pattern used before, do you know of any docs or examples that say/use ternary true/false is used over `!my_condition`?

**Comment by johnpmitsch** [comment]
Functionality works well!

One suggestions for improvement:  katello-service status spits out a lot of info, grepping for "dead" currently won't show anywhere else, but there is chace some other output in the command could show "dead" and give a false postivie. Is there something more specific we can use? i.e. ` inactive \(dead\) `?

File: packages/katello/katello/backup.rb (hunk 1 above)
Comment (cfouant) [reply packages/katello/katello/backup.rb]: I'm happy to change it, I prefer it that way. I recall it from the katello dev guide a while back, but no longer see it. Might have been a different style guide altogether.

**Comment by johnpmitsch** [comment]
When services are stopped and they are started by the scripts, there is no feedback to the user that this is happening, which can be confusing.

```
[root@three-six foreman-packaging]# ./packages/katello//katello/katello-backup --online /backup/
Starting backup: 2018-04-11 14:41:15 +0000
Creating backup folder /backup/katello-backup-20180411144115
### there is a long pause here, which makes one think its taking a long time to create a folder and something may have hung  ###
Generating metadata ... 
...
```
Can we print "Starting Services" and "Stopping Services" in their respective methods?

**Comment by cfouant** [comment]
@johnpmitsch - addition made, thanks!

**Comment by johnpmitsch** [comment]
@mbacovsky this was merged FYI

**Pull request merged**